device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/amqps/AmqpsIotHubConnection.java

```diff
@@ -468,9 +470,29 @@ public void onAuthenticationSessionOpened()
     }
 
     @Override
-    public void onMessageAcknowledged(Message message)
+    public void onMessageAcknowledged(Message message, DeliveryState deliveryState)
     {
-        this.listener.onMessageSent(message, null);
+        if (deliveryState == Accepted.getInstance())
+        {
+            this.listener.onMessageSent(message, null);
+        }
+        else if (deliveryState instanceof Rejected)
+        {
+            // The message was not accepted by the server, and the reason why is found within the nested error
+            TransportException ex = AmqpsExceptionTranslator.convertFromAmqpException(((Rejected) deliveryState).getError());
+            this.listener.onMessageSent(message, ex);
```

Comment:
The exception translator above does the hard work of figuring out if the message can be retried or not